src/lib/Output/Generator/Json/FieldTypeHashGenerator.php

```diff
@@ -32,24 +51,37 @@ public function generateHashValue($parent, $hashElementName, $hashValue)
      */
     protected function generateValue($parent, $value)
     {
-        switch (($hashValueType = gettype($value))) {
-            case 'NULL':
-            case 'boolean':
-            case 'integer':
-            case 'double':
-            case 'string':
-
-                // Will be handled accordingly on serialization
-                return $value;
-                break;
-
-            case 'array':
-                return $this->generateArrayValue($parent, $value);
-                break;
-
-            default:
-                throw new \Exception('Invalid type in Field value hash: ' . $hashValueType);
+        if ($value === null || is_scalar($value)) {
+            // Will be handled accordingly on serialization
+            return $value;
+        }
+
+        if (is_array($value)) {
+            return $this->generateArrayValue($parent, $value);
         }
+
+        if (is_object($value)) {
+            try {
+                $value = $this->normalizer->normalize($value, 'json', ['parent' => $parent]);
+            } catch (ExceptionInterface $e) {
+                $message = sprintf(
+                    'Unable to normalize value for type "%s". %s. '
+                    . 'Ensure that a normalizer is registered with tag: "%s".',
+                    get_class($value),
+                    $e->getMessage(),
+                    'ibexa.rest.serializer.normalizer',
+                );
+                $this->logger->error($message, [
+                    'exception' => $e,
+                ]);
+
+                $value = null;
+            }
+
+            return $this->generateValue($parent, $value);
+        }
+
+        throw new \Exception('Invalid type in Field value hash: ' . get_debug_type($value));
```

Comment:
Can we have here early throw for `if (!is_object ...)` to avoid too much nesting?

Reply:
I could do that, but as you can see there are two preceeding conditions: `is_array($value)` and `$value === null || is_scalar($value)`. Combining "normal" logic expressions and negated ones requires you to switch your "mental model" of the flow, so to speak.

Currently the flow is:
1. Is it a scalar or null?
2. Is it an array?
3. Is it an object?
4. Throw exception.

With what you're proposing we would have:
1. Is a scalar or null?
2. Is it an array?
3. Is it NOT an object? (Throw exception)
4. Handle object.

I think the above shows that we should either use early returns everywhere here, or not at all. WDYT?